examples/datasets/sentiment_descriptiveness.py

```diff
@@ -0,0 +1,189 @@
+}
+
+
+def deduplicate_query(ds):
+    query = set()
+    ranges = []
+    for i in range(len(ds)):
+        query_str = str(ds[i]["query"])
+        if query_str not in query:
+            query.add(query_str)
+            ranges.append(i)
+    return ds.select(ranges)
+
+
+if __name__ == "__main__":
+    args = HfArgumentParser(ScriptArguments).parse_args_into_dataclasses()[0]
+    if args.hf_entity is None:
+        args.hf_entity = api.whoami()["name"]
+    full_repo_id = f"{args.hf_entity}/{args.hf_repo_id}"
+
+    model_name = "gpt2"
+    dataset_tokenizer = AutoTokenizer.from_pretrained("gpt2")  # of the dataset
+
+    ################
+    # Dataset
+    ################
+    json = hf_hub_download(
+        repo_id="vwxyzjn/lm-human-preferences",
+        repo_type="dataset",
+        filename=task_to_filename[args.task],
+    )
+
+    MAGIC_TRAIN_NUMBER = 4992  # taken from https://github.com/openai/lm-human-preferences/blob/cbfd210bb8b08f6bc5c26878c10984b90f516c66/launch.py#L70
+    individual_ds = Dataset.from_json(json)
+    individual_ds = deduplicate_query(individual_ds)
+    ds = DatasetDict(
+        {
+            "train": individual_ds.select(range(MAGIC_TRAIN_NUMBER)),
+            "test": individual_ds.select(range(MAGIC_TRAIN_NUMBER, len(individual_ds))),
+        }
+    )
+
+    MAX_DEBUG_SAMPLES = 50
+    if args.debug:
+        for key in ds:
+            ds[key] = ds[key].select(range(min(MAX_DEBUG_SAMPLES, len(ds[key]))))
+
+    # columns are `['sample2', 'sample3', 'sample0', 'query', 'sample1', 'best']`
+    NUM_SAMPLES = 4
+
+    # edge cases handling: remove the cases where all samples are the same
+    def filter(row):
+        best_idx = row["best"]
+        chosen_sample = row[f"sample{best_idx}"]
+        if all(chosen_sample == row[f"sample{j}"] for j in range(NUM_SAMPLES)):
+            return False
+        else:
+            return True
+
+    print("=== Before filtering ===", ds)
+    ds = ds.filter(filter, load_from_cache_file=False)
+    print("=== After filtering ===", ds)
+
+    # here we simply take the preferred sample as the chosen one and the first non-preferred sample as the rejected one
+    def process(row):
+        for j in range(NUM_SAMPLES):
+            row[f"sample{j}"] = dataset_tokenizer.batch_decode(row[f"sample{j}"])
+        row["prompt"] = dataset_tokenizer.batch_decode(row["query"])
+        row["prompt"] = [item.strip() for item in row["prompt"]]
+        row["chosen"] = []
+        row["rejected"] = []
+        for i in range(len(row["best"])):
+            best_idx = row["best"][i]
+            chosen_sample = row[f"sample{best_idx}"][i].strip()
+            row["chosen"].append(
+                [
+                    {"role": "user", "content": row["prompt"][i].strip()},
+                    {"role": "assistant", "content": chosen_sample},
+                ]
+            )
+            # find the first rejected sample which is different from the chosen one
```

Comment:
Note that this is tricky because for some examples we do see some completions being the same.